snuba/datasets/plans/single_storage.py

```diff
@@ -133,11 +142,22 @@ def __init__(
 
     @with_span()
     def build_plan(self, request: Request) -> ClickhouseQueryPlan:
-        storage, mappers = self.__selector.select_storage(
-            request.query, request.settings
-        )
-        clickhouse_query = QueryTranslator(mappers).translate(request.query)
-        clickhouse_query.set_from_clause(storage.get_schema().get_data_source())
+        with sentry_sdk.start_span(
+            op="build_plan.selected_storage", description="select_storage"
+        ):
+            storage, mappers = self.__selector.select_storage(
+                request.query, request.settings
+            )
+
+        with sentry_sdk.start_span(
+            op="build_plan.selected_storage", description="translate"
+        ):
+            clickhouse_query = QueryTranslator(mappers).translate(request.query)
```

Comment:
Could you please add a comment here and in QueryTranslator explaining not to reuse this instance and why ?